packages/design-system-react-native/src/components/AvatarIcon/AvatarIcon.figma.tsx

```diff
@@ -0,0 +1,43 @@
+// import figma needs to remain as figma otherwise it breaks code connect
+// eslint-disable-next-line import-x/no-named-as-default
+import figma from '@figma/code-connect';
+import React from 'react';
+
+import { IconName } from '../Icon';
+
+import { AvatarIcon } from './AvatarIcon';
+
+import { AvatarIconSize, AvatarIconSeverity } from '.';
+
+/**
+ * -- This file was auto-generated by Code Connect --
+ * React Native implementation of AvatarIcon component
+ * `props` includes a mapping from Figma properties and variants to
+ * suggested values. You should update this to match the props of your
+ * code component, and update the `example` function to return the
+ * code example you'd like to see in Figma
+ */
+
+figma.connect(
+  AvatarIcon,
+  'https://www.figma.com/design/1D6tnzXqWgnUC3spaAOELN/%F0%9F%A6%8A-WIP--MMDS-Components?node-id=224%3A1636',
+  {
+    props: {
+      size: figma.enum('size', {
+        Xs: AvatarIconSize.Xs,
+        Sm: AvatarIconSize.Sm,
+        Md: AvatarIconSize.Md,
+        Lg: AvatarIconSize.Lg,
+        Xl: AvatarIconSize.Xl,
+      }),
+      severity: figma.enum('severity', {
+        neutral: AvatarIconSeverity.Neutral,
+        info: AvatarIconSeverity.Info,
+        success: AvatarIconSeverity.Success,
+        error: AvatarIconSeverity.Error,
+        warning: AvatarIconSeverity.Warning,
+      }),
+    },
+    example: (props) => <AvatarIcon iconName={IconName.Wallet} {...props} />,
```

Comment:
iconName prop is quite complicated as we have a lot of icons. Will come back on how to best handle this.